src/test_util.rs

```diff
@@ -151,6 +151,16 @@ pub fn collect_n<T>(mut rx: mpsc::Receiver<T>, n: usize) -> impl Future<Item = V
     })
 }
 
+pub fn is_empty<T>(mut rx: mpsc::Receiver<T>) -> impl Future<Item = bool, Error = ()> {
```

Comment:
not sure why we need this when you can just call `rx.poll()?.is_not_ready()`?

Reply:
Good point